spreadsheets.js

```diff
@@ -0,0 +1,215 @@
+        round_end.subtract(options.offset_hours, 'hours');
+    }
+    return [round_start, round_end];
+}
+
+function find_pairing(service_account_auth, spreadsheet_key, white, black, callback) {
+    var doc = new GoogleSpreadsheet(spreadsheet_key);
+    doc.useServiceAccountAuth(service_account_auth, function(err, info) {
+        var pairings_sheet = undefined;
+        doc.getInfo(function(err, info) {
+            if (err) { return callback(err, info); }
+            // Find the last spreadsheet with the word "round" in the title.
+            // this ought to work for both tournaments
+            info.worksheets.forEach(function(sheet) {
+                if (sheet.title.toLowerCase().indexOf("round") != -1) {
+                    pairings_sheet = sheet;
+                }
+            });
+            if (!pairings_sheet) {
+                callback("Unable to find pairings worksheet");
+                return;
+            }
+            // Query for the appropriate row.
+            // Again, this ought to work for both tournaments
+            pairings_sheet.getRows({
+                offset: 1,
+                limit: 100
+            }, function(err, rows) {
+                if (err) { return callback(err, rows); }
+                var potential_rows = [];
+                rows.forEach(function(row) {
+                    var row_black = row.black.toLowerCase();
+                    var row_white = row.white.toLowerCase();
+                    if (
+                        (row_black.indexOf(black) != -1 && row_white.indexOf(white) != -1)
+                    ) {
+                        potential_rows.push([row, false]);
+                    } else if (
+                        (row_white.indexOf(black) != -1 && row_black.indexOf(white) != -1)
+                    ) {
+                        potential_rows.push([row, true]);
+                    }
+                });
+                // Only update it if we found an exact match
+                if (potential_rows.length > 1) {
+                    return callback("Unable to find pairing. More than one row was returned!");
+                } else if(potential_rows.length < 1) {
+                    return callback("Unable to find pairing. No rows were returned!");
+                }
+                var row = potential_rows[0][0];
+                var reversed = potential_rows[0][1];
+                callback(undefined, row, reversed);
+            });
+        });
+    });
+
+}
+
+// Update the schedule
+function update_schedule(service_account_auth, key, colname, format, schedule, callback) {
+    var white = schedule.white;
+    var black = schedule.black;
+    var date = schedule.date;
+    find_pairing(service_account_auth, key, white, black, function(err, row, reversed) {
+        if (err) {
+            return callback(err);
+        }
+        // This portion is specific to the team tournament
+        row[colname] = date.format(format);
```

Comment:
cell by column name is nice. glad this is part of the api. much better than my hard coded indices. ill have to remove them and rewrite my stuff to use the row interface where possible.